libtiny_tui/src/text_field.rs

```diff
@@ -9,7 +9,8 @@ use termbox_simple::Termbox;
 use crate::{config::Colors, termbox, trie::Trie, utils, widget::WidgetRet};
 
 // TODO: Make these settings
-const SCROLLOFF: i32 = 5;
+const SCROLL_OFF: i32 = 5;
+const OVERFLOW_WIDTH: i32 = 36;
```

Comment:
I don't understand why this is necessary. @trevarj could you document this please before I merge this PR?

Reply:
I will add some documentation comments and rename `OVERFLOW_WIDTH`

Reply:
Honestly, I don't even know what `SCROLL_OFF` does! I'm playing around with changing its value and it doesn't seem like it does anything.

Reply:
> I'm playing around with changing its value and it doesn't seem like it does anything.

Yeah I tried this as well. Also tried just deleting it, and nothing broke! Let's remove it.

Reply:
Ahh, you mean `SCROLL_OFF`. That's actually useful, I'll document it.

Reply:
Never mind, I see what it does, but I can't explain it in words!

Reply:
It's inspired by vim's [`scrolloff`](http://vimdoc.sourceforge.net/htmldoc/options.html#'scrolloff') ;-)

Reply:
About `OVERFLOW_WIDTH`, I renamed it to `SCROLL_FALLBACK_WIDTH`. It's the value that is used when you have `text_field_wrap=true` and then resize the terminal. When the text_field's width is < 36 it will fall back to scroll mode.

Reply:
Why is this necessary? If the terminal width is 20, with enough height it should be possible to use wrapping. Is there a problem with this?

I don't know if you've changed the code, but in the version I've reviewed I removed this variable (and the conditional) and everything worked as before -- I was able to use wrapping on a small terminal. So I'm also not sure if it works as intended.

Reply:
Oh, I just thought you wanted it to fall back to scrolling when the terminal is too small. 

From your first comment:

>     * On small screens we should fall back to the current behavior, which is more compact (always takes 1 row). When to fall back exactly? A few random ideas (feel free to suggest more):

Reply:
> I just thought you wanted it to fall back to scrolling when the terminal is too small.

I still do, but I'm not sure if this is the best way to do it.

I guess it's fine for now. I just tried again and it seems to work as advertised.